server/recipeAPI/spoonacularAPI/SpoonacularRecipeAPI.ts

```diff
@@ -0,0 +1,312 @@
+
+    protected recipeSearchParameters: Set<string>;
+    protected recipeGetParameters: Set<string>;
+
+    protected dishTypes: Set<string>;
+
+    protected headers: any;
+
+    constructor(foodAPI: IFoodAPI) {
+        this.foodAPI = foodAPI;
+
+        this.apiKey = process.env.SPOONACULAR_API_KEY;
+        this.host = process.env.SPOONACULAR_HOST;
+
+        this.dishTypes = new Set([
+            "main course",
+            "side dish",
+            "dessert",
+            "appetizer",
+            "salad",
+            "bread",
+            "breakfast",
+            "soup",
+            "beverage",
+            "sauce",
+            "drink"
+        ]);
+
+        this.recipeSearchParameters = new Set([
+            'query',
+            'cuisine',
+            'diet',
+            'intolerences',
+            'includeIngredients',
+            'type'
+        ]);
+
+        this.recipeGetParameters = new Set([
+
+        ]);
+    }
+
+    /**
+     * Retrieves collection of partially filled Recipe items that satisfy searching parameters.
+     * 
+     * @param parameters parameters used for searching.
+     * - query - required parameter that defines the name of the Recipe item (partial names are accepted).
+     * - number - optional parameter that defines max number of the results to be returned.
+     * - page - optional parameter that defines page number for pagination.
+     * - intolerence - optional parameter that defines the type of intolerences to be taken in consideration during searching.
+     * - cuisine - optional parameter that limits search results to specific cuisines.
+     * - type - optional parameter that limits search results to specific dish types.
+     * 
+     * @throws NoParameterFound exception when required parameters weren't found.
+     * @throws ParameterIsNotAllowed exception when encountering a non-existing parameter.
+     * 
+     * @returns Promise filled with a collection of Partial<IRecipe> objects.
+     */
+    async SearchRecipe(parameters: Map<string, any>): Promise<Partial<IRecipe>[]> {
+        let searchRecipeURL = process.env.SPOONACULAR_RECIPE_BASE_URL + '/complexSearch';
+
+        let urlSearchParameters = this.convertSearchRecipeParameters(parameters);
+
+        let response = await axios(
+            searchRecipeURL,
+            {
+                headers: this.headers,
+                params: urlSearchParameters
+            }
+        );
+
+        let jsonArray: any[] = response.data.results;
+        let recipeArray: IRecipe[] = [];
+
+        for (let i = 0; i < jsonArray.length; i++) {
+            recipeArray.push(
+                await this.parseRecipe(jsonArray[i])
+            );
+        }
+
+        return recipeArray;
+    }
+
+    /**
+     * Converts search recipe parameters to URLSearchParams.
+     * 
+     * @param parameters parameters used for searching.
+     * - query - required parameter that defines the name of the Recipe item (partial names are accepted).
```

Comment:
`recipeName`

Reply:
#50 will be implemented in future branches